pkg/api/v1/types.go

```diff
@@ -1961,13 +1961,30 @@ type PodLogOptions struct {
 	// The container for which to stream logs. Defaults to only container if there is one container in the pod.
 	Container string `json:"container,omitempty"`
 
-	// Follow the log stream of the pod.
-	// Defaults to false.
+	// Follow the log stream of the pod. Defaults to false.
 	Follow bool `json:"follow,omitempty"`
 
-	// Return previous terminated container logs.
-	// Defaults to false.
+	// Return previous terminated container logs. Defaults to false.
 	Previous bool `json:"previous,omitempty"`
+
+	// An RFC3339 timestamp or a relative time duration relative to the current time
+	// from which to show logs. If this value precedes the time a pod was started, only
+	// logs since the pod start will be returned. If this value is in the future, no logs
+	// will be returned.
+	Since string `json:"since,omitempty"`
+
+	// If true, add an RFC3339 or RFC3339Nano timestamp at the beginning of every line
+	// of log output. Defaults to false.
+	Timestamps bool `json:"timestamps,omitempty"`
+
+	// If set, the number of lines from the end of the logs to show. If not specified,
+	// logs are shown from the since parameter or pod creation, whichever is earlier.
```

Comment:
As CLI args: `since-duration`?  `since-relative`?  We don't use a duration
string in the API yet, but it's a very useful value for clients.
Preference?

On Mon, Sep 14, 2015 at 2:00 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/13780#discussion_r39425341>
> :
>
> >  	Previous bool `json:"previous,omitempty"`
> > +
> > +	// An RFC3339 timestamp or a relative time duration relative to the current time
> > +	// from which to show logs. If this value precedes the time a pod was started, only
> > +	// logs since the pod start will be returned. If this value is in the future, no logs
> > +	// will be returned.
> > +	Since string `json:"since,omitempty"`
> > +
> > +	// If true, add an RFC3339 or RFC3339Nano timestamp at the beginning of every line
> > +	// of log output. Defaults to false.
> > +	Timestamps bool `json:"timestamps,omitempty"`
> > +
> > +	// If set, the number of lines from the end of the logs to show. If not specified,
> > +	// logs are shown from the since parameter or pod creation, whichever is earlier.
>
> Doesn't later make more sense?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13780/files#r39425341>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift